plugin-gradle/src/main/java/com/diffplug/gradle/spotless/SpotlessTaskImpl.java

```diff
@@ -94,26 +94,34 @@ public void performAction(InputChanges inputs) throws Exception {
 
 	private void processInputFile(@Nullable GitRatchet ratchet, Formatter formatter, File input) throws IOException {
 		File output = getOutputFile(input);
-		getLogger().debug("Applying format to " + input + " and writing to " + output);
+		getLogger().debug("Applying format to {} and writing to {}", input, output);
 		PaddedCell.DirtyState dirtyState;
 		if (ratchet != null && ratchet.isClean(getProjectDir().get().getAsFile(), getRootTreeSha(), input)) {
 			dirtyState = PaddedCell.isClean();
 		} else {
-			dirtyState = PaddedCell.calculateDirtyState(formatter, input);
+			try {
+				dirtyState = PaddedCell.calculateDirtyState(formatter, input);
+			} catch (IOException e) {
+				throw new IOException("Issue processing file: " + input, e);
+			} catch (RuntimeException e) {
+				throw new IllegalArgumentException("Issue processing file: " + input, e);
+			}
```

Comment:
I'm fine with this, but I'm curious why it is needed. Are you parsing these? Do we need a `SpotlessException` with a field `File`?

Reply:
Nope x2. The point is just to get the fileName in the Exception message. Added a unitTest

Reply:
FYI the test failure is real (Windows `\` vs unix `/`). Here's the method we've been using to detect win

https://github.com/diffplug/spotless/blob/f65d366d57cb66286a137827c09c6504d51414e3/lib/src/main/java/com/diffplug/spotless/FileSignature.java#L112-L115

Reply:
Thanks. I updated the test